### PR TITLE
login-page

### PR DESCRIPTION
**Pull request by theVivek-kumar**
logIn page
SignIn page

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -247,19 +276,27 @@ color: var(--PRIMERY-COLOR);
...
+  overflow: scroll;
+}
 .main-section-wrapper{
 display: grid;
 grid-template-columns: 17rem 1fr;
 }
 .duration-label {
     position: relative;
+    width: 6rem;
     background-color: #222222d8;
```
Comment (Adarsh-Patel-Dev) [inline src/App.css:293]: use color variable here

File: src/App.js [hunk 2]
```diff
@@ -2,12 +2,17 @@ import "./App.css";
...
+import './components/SignIn/signIn.css'
 import LandingPage from "./page/landingPage";
 import SingleVideoPage from "./page/SingleVideoPage";
 import VideoListing from "./page/videoListing";
 import WatchLaterpage from "./page/WatchLaterpage";
 import  HistoryPage  from "./page/History-page";
 import LikeVideoPage from "./page/LikeVideoPage";
+import SignIn from "./components/SignIn/SignIn";
+import Login from "./components/login/Login";
+import PlayList from "./components/PlayList";
```
Comment (Adarsh-Patel-Dev) [inline src/App.js:15]: You can export all components and pages into their respective index.js file and then import from  that file.

File: src/components/MainArea.js [hunk 3]
```diff
@@ -54,10 +55,96 @@ const videourl = `https://www.youtube.com/embed/${video_id}`;
...
+                <RiBarChartHorizontalLine />  short by
+              </div>
+              </div>
+            </div>
+            <div>
+           <div className='chennel-container' ><img className='chennel-logo-img' src={creater_img} ></img> <div>{ creator}</div> </div>  
+             <p className='comment-md'>You can also try turning off any adblockers you may have running. These sometimes interfere with comments and block them from loading.</p>
+           
+             <div className="single-video-icons">
+                <AiFillLike />
```
Comment (Adarsh-Patel-Dev) [inline src/components/MainArea.js:72]: Add addtolikevideo functionality on `  <AiFillLike />` icon

File: src/components/SignIn/SignIn.css [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    height: auto;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
+.login-container-form{
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
```
Comment (Adarsh-Patel-Dev) [inline src/components/SignIn/SignIn.css:103]: you can create utility class for centering like `flex-center-column` & reuse it
`.flex-center-column{
display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    }
`

File: src/components/login/login.css [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    font-weight: 500;
+    font-size: 1.3rem;
+    margin-top: 1rem;
+}
+
+
+.label {
+    font-weight: 500;
+    display: flex;
+    color: var(--GRAY-90)!important;
```
Comment (Adarsh-Patel-Dev) [inline src/components/login/login.css:54]: try to avoid `!important` in your code

File: src/components/CardHorizontal.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    console.log(_id)
+    const navigate = useNavigate();
+    const { addToHistory, setHistory } = useHistoryContext();
+    const { removeFromWatchLater,setWatchLater } = useWatchLaterContext();
+    return (
+        <>
+            <div className='card-horizontal-container'>
+                <div className='card-horizontal image-md'>
+                    <img onClick={() => {
+                addToHistory(videos, setHistory);
```
Comment (m26unkwn) [inline src/components/CardHorizontal.js:27]: write this function outside of the jsx.

File: src/context/PlayListContext.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    setDisplay(display === "none" ? "block" : "none");
+  };
+
+  async function getPlaylists(setPlaylist) {
+    try {
+      const response = await axios({
+        method: "GET",
+        url: "/api/user/playlists",
+        headers: { authorization: localStorage.getItem("token") },
+      });
```
Comment (m26unkwn) [inline src/context/PlayListContext.js:23]: extract this token outside of the component than use it here.

**Review by m26unkwn — COMMENTED** [review]
Abstraction of your code is good
you can go ahead with code.

**Pull request merged**